src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -505,6 +513,9 @@
     <Field Name="Alternative" Type="BoundExpression"/>
 
     <Field Name="ConstantValueOpt" Type="ConstantValue?"/>
+
+    <Field Name="NaturalTypeOpt" Type="TypeSymbol?" Null="allow"/>
+    <Field Name="WasTargetTyped" Type="bool" />
```

Comment:
I'm going to use this changes as a baseline for new() so we're consistent over all target-typing features.

Reply:
I'll probably rebase this PR.  It is still under development.